src/frontend/src/optimizer/plan_node/generic/insert.rs

```diff
@@ -60,6 +61,24 @@ impl<PlanRef: GenericPlanRef> Insert<PlanRef> {
         }
     }
 
+    fn logical_pk(&self) -> Option<Vec<usize>> {
+        None
+    }
+
+    fn ctx(&self) -> OptimizerContextRef {
+        self.input.ctx()
+    }
+}
+
+impl<PlanRef: GenericPlanRef> Insert<PlanRef> {
+    pub fn output_len(&self) -> usize {
+        if self.returning {
+            self.input.schema().len()
+        } else {
+            1
+        }
+    }
```

Comment:
I think `ouput_len` should be `self.schema.len()` rather than `self.input.schema().len()`?

Reply:
+1

Reply:
fixed